### PR TITLE
Minor change in order to fix the switch status

### LINKED ISSUE #3: Tuya switch component works, but the toggle does not keep its state
Hello. I'm merely mirroring the issue [#25992](https://github.com/home-assistant/home-assistant/issues/25992) that i posted on the home-assistant's issues as i think linking between them might catch more attraction to this.

Also, it appears that i'm not the only one affected by this according to [this post](https://community.home-assistant.io/t/is-tuya-broken/104006/28) on HA community.

EDIT: Ooops. I totally forgot to thank you for your work! -- what a jackass i am!

**--- BEGIN COPY ---**    

**Home Assistant release with the issue:**
0.97.2 

**Last working Home Assistant release (if known):**
Unknown.

**Operating environment (Hass.io/Docker/Windows/etc.):**
Ubuntu, 18.04.

**Component/platform:**
https://www.home-assistant.io/components/tuya/

**Description of problem:**
The switching on/off works, but the state of the toggle gets reset (eg: I turn on the light, the light gets turned on, but the toggle reverts to the off state)

**Problem-relevant `configuration.yaml` entries and (fill out even if it seems unimportant):**
```yaml
tuya:
  username: phone_number
  password: password
  country_code: countrycode
  platform: smart_life

```

**Traceback (if applicable):**
none.

**Additional information:**

No errors in any logs.

### PR DESCRIPTION
**Pull request by niladam**
This could and **probably** should be a single function to get the status of the device based on it's specified id.

I quickfixed this in order for the switches to function, and i think 2 weeks is a pretty long time. Also, please keep in mind that my python language skills are from none to low :)

I think if this gets merged it'll be a really great and really quick fix :)

This should fix #3

### CODE REVIEW

**Comment by niladam** [comment]
Personally, i think there could an easier way of doing this, without actually requesting the devices again from the API. However, i'm not really sure if this could actually work, since the SESSION basically holds the state -- which upon triggering changes -- therefore the SESSION data could become stale.

@PaulAnnekov - thoughts about this buddy?

**Comment by PaulAnnekov** [comment]
This code looks a bit dirty, but as I have no time to handle it better I will merge this PR. Anyway, thanks :wink: .

**Comment by niladam** [comment]
@PaulAnnekov 

> This code looks a bit dirty, but as I have no time to handle it better I will merge this PR. Anyway, thanks 😉 .

I know! I'm not a python developer and i think this is the 2nd time i'm doing something with it :P - and obviously, i already stated that about my skills.

I'm curious though, how you would've done the foreach ?

File: tuyaha/devices/switch.py [hunk 1]
```diff
@@ -1,10 +1,14 @@
 from tuyaha.devices.base import TuyaDevice
 
-
 class TuyaSwitch(TuyaDevice):
 
     def state(self):
-        state = self.data.get('state')
+        devices = self.api.discover_devices()
```
Comment (MartinHjelmare) [inline tuyaha/devices/switch.py]: This does a request which now causes home assistant event loop to block.

Home assistant needs to be updated to not call this method `state` in the entity `is_on` property. It should be called in `update` entity method instead and cache the result.

**Pull request merged**